controllers/cinder_controller.go

```diff
@@ -890,22 +880,12 @@ func (r *CinderReconciler) apiDeploymentCreateOrUpdate(ctx context.Context, inst
 
 func (r *CinderReconciler) schedulerDeploymentCreateOrUpdate(ctx context.Context, instance *cinderv1beta1.Cinder) (*cinderv1beta1.CinderScheduler, controllerutil.OperationResult, error) {
 
-	svcTemplate := instance.Spec.CinderScheduler
 	cinderSchedulerSpec := cinderv1beta1.CinderSchedulerSpec{
-		ContainerImage:             svcTemplate.ContainerImage,
-		Replicas:                   svcTemplate.Replicas,
-		Debug:                      svcTemplate.Debug,
-		CustomServiceConfig:        svcTemplate.CustomServiceConfig,
-		CustomServiceConfigSecrets: svcTemplate.CustomServiceConfigSecrets,
-		DefaultConfigOverwrite:     svcTemplate.DefaultConfigOverwrite,
-		NetworkAttachments:         svcTemplate.NetworkAttachments,
-		ServiceUser:                instance.Spec.ServiceUser,
-		DatabaseHostname:           instance.Status.DatabaseHostname,
-		DatabaseUser:               instance.Spec.DatabaseUser,
-		Secret:                     instance.Spec.Secret,
-		ExtraMounts:                instance.Spec.ExtraMounts,
-		TransportURLSecret:         instance.Status.TransportURLSecret,
-		PasswordSelectors:          instance.Spec.PasswordSelectors,
+		CinderTemplate:          instance.Spec.CinderTemplate,
+		CinderSchedulerTemplate: instance.Spec.CinderScheduler,
+		ExtraMounts:             instance.Spec.ExtraMounts,
+		DatabaseHostname:        instance.Status.DatabaseHostname,
```

Comment:
I was concerned about this field at the beginning, because it comes from the upper layer `Status`, but I see you still explicitly set it here.